test/vegeta/pacers.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// pacers contains self-defined pacers that can be used in Attack of Vegeta (for performance testing)
+
+package vegeta
+
+import (
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	vegeta "github.com/tsenart/vegeta/lib"
+)
+
+// steadyUpPacer is a Pacer that describes attack request rates that increases in the beginning then becomes steady.
+//  Max  |     ,----------------
+//       |    /
+//       |   /
+//       |  /
+//       | /
+//  Min -+------------------------------> t
+//       |<-Up->|
+type steadyUpPacer struct {
+	// upDuration is the duration that attack request rates increase from Min to Max, must be larger than 0.
+	upDuration time.Duration
+	// min is the attack request rates from the beginning, must be larger than 0.
+	min vegeta.Rate
+	// max is the maximum and final steady attack request rates, must be larger than Min.
+	max vegeta.Rate
+
+	slope        float64
+	minHitsPerNs float64
+	maxHitsPerNs float64
+}
+
+// NewSteadyUpPacer returns a new SteadyUpPacer with the given config.
+func NewSteadyUpPacer(min vegeta.Rate, max vegeta.Rate, upDuration time.Duration) vegeta.Pacer {
```

Comment:
Can we add test coverage for these?

Reply:
Done.
And below are the actual QPS curves for these two Pacers drawn from Mako:
SteadyUpPacer: https://mako.dev/run?run_key=6524866857533440&~l=1
CombinedPacer: https://mako.dev/run?run_key=5425050018643968&~l=1